src/codegeneration/CommonJsModuleTransformer.js

```diff
@@ -83,7 +83,7 @@ export class CommonJsModuleTransformer extends ModuleTransformer {
     // "module.exports", so we don't append "module.exports = {}" to the output.
     if (this.hasExports()) {
       var descriptors = this.transformObjectLiteralToDescriptors(exportObject);
-      var exportStatement = parseStatement `Object.defineProperties(exports, ${descriptors});`;
+      var exportStatement = parseStatement `Object.defineProperties(module.exports, ${descriptors});`;
       statements = prependStatements(statements, exportStatement);
     }
```

Comment:
If it doesn't have `module.exports` or `require` then it is a CommonJS file with no exports or dependencies. In such a scenario, treating it as a global is equivalent, so the detection can fail fine.